core/src/main/java/com/linecorp/armeria/client/ClientBuilder.java

```diff
@@ -67,96 +66,43 @@
-        this(requireNonNull(uri, "uri"), null, null, null);
-    }
-
-    /**
-     * Creates a new {@link ClientBuilder} that builds the client that connects to the specified
-     * {@link Endpoint} with the {@code scheme}.
-     *
-     * @deprecated Use {@link Clients#builder(String, EndpointGroup)}.
-     */
-    @Deprecated
-    public ClientBuilder(String scheme, Endpoint endpoint) {
-        this(Scheme.parse(requireNonNull(scheme, "scheme")), requireNonNull(endpoint, "endpoint"));
-    }
-
-    /**
-     * Creates a new {@link ClientBuilder} that builds the client that connects to the specified
-     * {@link Endpoint} with the {@link Scheme}.
-     *
-     * @deprecated Use {@link Clients#builder(Scheme, EndpointGroup)}.
-     */
-    @Deprecated
-    public ClientBuilder(Scheme scheme, Endpoint endpoint) {
-        this(null, requireNonNull(scheme, "scheme"), null, requireNonNull(endpoint, "endpoint"));
-    }
+    private final String path;
+    private Scheme scheme;
+    private boolean updatedScheme;
 
-    /**
-     * Creates a new {@link ClientBuilder} that builds the client that connects to the specified
-     * {@link Endpoint} with the {@link SessionProtocol}.
-     *
-     * @deprecated Use {@link Clients#builder(SessionProtocol, EndpointGroup)}.
-     */
-    @Deprecated
-    public ClientBuilder(SessionProtocol protocol, Endpoint endpoint) {
-        this(null, null, requireNonNull(protocol, "protocol"), requireNonNull(endpoint, "endpoint"));
+    ClientBuilder(URI uri) {
+        checkArgument(uri.getScheme() != null, "uri must have scheme: %s", uri);
+        checkArgument(uri.getRawAuthority() != null, "uri must have authority: %s", uri);
+        this.uri = uri;
+        endpointGroup = null;
+        path = null;
+        scheme = Scheme.parse(uri.getScheme());
     }
 
-    ClientBuilder(@Nullable URI uri, @Nullable Scheme scheme, @Nullable SessionProtocol protocol,
-                  @Nullable EndpointGroup endpointGroup) {
-        this.uri = uri;
-        this.scheme = scheme;
-        this.protocol = protocol;
+    ClientBuilder(Scheme scheme, EndpointGroup endpointGroup, @Nullable String path) {
+        if (path != null) {
+            checkArgument(path.startsWith("/"),
+                          "path: %s (expected: an absolute path starting with '/')", path);
+        }
+        uri = null;
         this.endpointGroup = endpointGroup;
+        this.path = path;
+        this.scheme = scheme;
     }
 
     /**
-     * Sets the {@code path} of the client.
+     * Sets the {@link SerializationFormat} of the client from the specified {@code format}.
      */
-    public ClientBuilder path(String path) {
-        ensureEndpointGroup();
-        requireNonNull(path, "path");
-        checkArgument(path.startsWith("/"), "path: %s (expected: an absolute path starting with '/')", path);
-        this.path = path;
-        return this;
+    public ClientBuilder serializationFormat(String format) {
```

Comment:
Similar to path, shouldn't this also go away since scheme is a required parameter?

Reply:
+1 if possible.

Reply:
Done